python/device/device.py

```diff
@@ -921,6 +958,10 @@ def test_twin_properties_thread(self, worker_thread_info):
                 continue
 
             else:
+                self.subtest_method_invoke()
+                worker_thread_info.watchdog_epochtime = time.time()
+                time.sleep(self.config[Settings.TWIN_UPDATE_INTERVAL_IN_SECONDS])
+
```

Comment:
Since there are are several different tests on this one thread, we might start running into issues there the tests start blocking each other too much.

You said that you are going to change how you do threading, so maybe you are already thinking about this.

Reply:
Agree 100%.  that's part of my thinknig for the refactor.